src/components/MainNav/MainNav.js

```diff
@@ -55,12 +60,20 @@ class MainNav extends Component {
     this.toggleUserMenu = this.toggleUserMenu.bind(this);
     this.logout = this.logout.bind(this);
     this.lastVisited = {};
+    this.unsub = null;
+    this.queryValues = null;
 
     const moduleList = modules.app.concat({
       route: '/settings',
       module: '@folio/x_settings',
     });
 
+    for (const entry of moduleList) {
+      if (props.location.pathname.startsWith(entry.route)) {
```

Comment:
Is this a precise enough test? What if `entry.route` is "foo" and the location is "/foobar"?